### ISSUE TITLE
ThemisElixir

### ISSUE THREAD

**Issue by WolfDan** [cossacklabs/themis#181]
Hi! 

I'd like to know if it's possible bring support for elixir lang (or Phoenix framework), it can be awesome 

Greetings

**Comment by gene-eu** [comment]
Hi,

we've assessed the possibility to build Themis for Erlang / Elixir support via NIFs (ports/porcelain being very expensive alternative performance-wise), but realized that it involves certain risk for Erlang users: 

- fault-tolerance of Erlang processes might be affected by the quality of NIF-surrounding code. While we're confident in Themis, we're not Erlang connoisseurs enough to ensure that users won't crash Themis from Erlang/Elixir land. And the price for error? You segfault Themis, Themis will segfault the whole Erlang VM. Even though in long-time production running instances of Themis we're not aware of segfaults for really long time now, we can't guarantee users won't cause one.
- it's worth noting that NIFs are not managed by the Erlang VM scheduler, and are executed in a single chunk of time, blocking things, so the performance quirks might sometimes be really upsetting. We don't quite understand the repercussions, or how to work around them, and we don't have the competence ourselves in Erlang VM to think responsibly of these matters. 

Moreover, we're not running Erlang code anywhere in our environments (yet), so that's not a priority for us, as of now.

**TL;DR**: We won't do that ourselves without significant Erlang VM competence on board. But we will be happy to assist someone from Erlang/Elixir land to build the wrapper*, and help work around some of the issues from our side. 

* - it's fairly easy, by the way, for some platforms it took less than 30 hours to do so.
